### PR TITLE
Added repository filter to global and org commands

### PR DESCRIPTION
**Pull request by acraven**
We would like to add the ability to filter repos by regexes. We have added the --includerepos and --excluderepos arguments to the global and org commands.

We would appreciate your feedback.

### CODE REVIEW

**Comment by AnthonySteele** [comment]
it looks sound. Sorry that my PR earlier today conflicts. Can you rebase?
These two new values are also good candidates to store in the settings file.

**Comment by acraven** [comment]
Thanks for the quick response. I have fixed the merge conflicts.
I don't really have an opinion on the -ir/er short options, we just use the long options for clarity.
We will take a look at the settings file before continuing.

**Comment by acraven** [comment]
ps. Would you be happy to Squash when accepting, or would you like us to squash?

File: NuKeeper/Engine/GitHubRepositoryDiscovery.cs [hunk 1]
```diff
@@ -24,10 +25,10 @@ public class GitHubRepositoryDiscovery : IGitHubRepositoryDiscovery
             switch (settings.Scope)
             {
                 case ServerScope.Global:
-                    return await ForAllOrgs(gitHub);
+                    return await ForAllOrgs(gitHub, settings);
 
                 case ServerScope.Organisation:
-                    return await FromOrganisation(gitHub, settings.OrganisationName);
+                    return await FromOrganisation(gitHub, settings.OrganisationName, settings);
```
[the remark this answers is not in the document]
Comment (acraven) [reply NuKeeper/Engine/GitHubRepositoryDiscovery.cs]: I did try that earlier, but I ran into problems with...
        var repos = await FromOrganisation(gitHub, org.Name ?? org.Login, settings);

I could add an optional argument to override OrganisationName in FromOrganisation.

[the remark this answers is not in the document]
Comment (AnthonySteele) [reply NuKeeper/Engine/GitHubRepositoryDiscovery.cs]: Ah right. That's correct then.

**Comment by acraven** [comment]
I've just checked that both -ir and -er work as expected. All good for you to accept and squash from our point of view.

**Comment by acraven** [comment]
Thanks v much. Would you expect us to update the wiki, or is that something you prefer to do yourself?

**Comment by AnthonySteele** [comment]
I will likely have time for that (and a release) this weekend.

**Comment by acraven** [comment]
Okay, cool. Have a good weekend.

**Comment by AnthonySteele** [comment]
No worries. I generally update the docs slightly after the release. That way they don't advertise features that aren't available yet.

**Pull request merged**